### PR TITLE
:wrench: chore: Document `OrganizationConfigIntegrationEndpoint`

### PR DESCRIPTION
**Pull request by iamrajjoshi**
image

i am privating `OrganizationConfigRepositoriesEndpoint` so we don't promote use of plugins.
i also added support for `providerKey` as the query param b/c its more standard for frontend.

### CODE REVIEW

File: src/sentry/integrations/api/endpoints/organization_config_integrations.py [hunk 1]
```diff
@@ -7,18 +10,49 @@
...
+from sentry.apidocs.utils import inline_sentry_response_serializer
+from sentry.integrations.api.serializers.models.integration import (
+    IntegrationProviderResponse,
+    IntegrationProviderSerializer,
+)
 from sentry.integrations.manager import default_manager as integrations
 
 
+class OrganizationConfigIntegrationsEndpointResponse(TypedDict):
+    providers: list[IntegrationProviderResponse]
```
Comment (iamrajjoshi) [inline src/sentry/integrations/api/endpoints/organization_config_integrations.py:25]: needed to this b/c our response is non standard. instead of just returning a list, we return a dict with 1 key "provider" which contains the list.

**Comment by iamrajjoshi** [comment]
> 🚨 **Warning:** This pull request contains Frontend and Backend changes!
> 
> It's discouraged to make changes to Sentry's Frontend and Backend in a single pull request. The Frontend and Backend are **not** atomically deployed. If the changes are interdependent of each other, they **must** be separated into two pull requests and be made forward or backwards compatible, such that the Backend or Frontend can be safely deployed independently.
> 
> Have questions? Please ask in the [`#discuss-dev-infra` channel](https://app.slack.com/client/T024ZCV9U/CTJL7358X).

i think this PR should be ok

File: api-docs/openapi.json [hunk 2]
```diff
@@ -141,7 +141,7 @@
     "/api/0/issues/{issue_id}/hashes/": {
       "$ref": "paths/events/issue-hashes.json"
     },
-    "/api/0/issues/{issue_id}/": {
+    "/api/0/issues/{organization_id_or_slug}/{issue_id}/": {
```
Comment (ameliahsu) [inline api-docs/openapi.json:144]: This should be `/api/0/organization/<organization_id_or_slug>/issues/<issue_id>/`, but maybe we should do this in a separate PR with all of the endpoints that need this update? I feel like these pages need a note at the top about the change

Comment (iamrajjoshi) [reply api-docs/openapi.json:144]: oh shit, this shouldn't have been updated, good catch 🙏🏽

File: src/sentry/api/endpoints/organization_config_repositories.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 class OrganizationConfigRepositoriesEndpoint(OrganizationEndpoint):
     owner = ApiOwner.INTEGRATIONS
     publish_status = {
-        "GET": ApiPublishStatus.UNKNOWN,
```
Comment (iamrajjoshi) [inline src/sentry/api/endpoints/organization_config_repositories.py:15]: don't want to promote plugin use

**Review by isabellaenriquez — APPROVED** [review]
Dino Well Done
